# Define the goal
float32[] goal                          # goal coordinates in 3d space
float32 max_time_taken                  # max time taken to reach the goal
float32 tolerance                       # when the distance to the goal falls within tolerance, the task is considered completed